From 554374e58ac130c0ef1cd45b65922af47a0ff550 Mon Sep 17 00:00:00 2001
From: Alexandre TORGUE <alexandre.torgue@st.com>
Date: Wed, 6 Mar 2024 11:53:45 +0100
Subject: [PATCH 603/954] feat(stm32mp25-fdts): update RIF configuration for DK
 board

Align the "open" configuration with the latest version of the xls tool.
It fixes the bl31_context area (start addr + size) and allows the CID0
to access to the "op-tee" area (as it is done for all area for a "open"
configuration).

It also changes PLL binding to satisfy YAML verification and moves
U(S)ART source clock to HSI_KER for low power feature.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git d9b3c28dde757f7608d5b5e4e3d7d133e62eb2d3 feat(stm32mp25-fdts): update RIF configuration for DK board]
Signed-off-by: Alexandre TORGUE <alexandre.torgue@st.com>
Change-Id: I553ea3011da787b9fa72451eb437c4c5e4d80611
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/380736
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Maxime MERE <maxime.mere@foss.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 fdts/stm32mp257f-dk-ca35tdcid-fw-config.dtsi | 18 +++++++++---------
 fdts/stm32mp257f-dk-ca35tdcid-rcc.dtsi       | 12 +++++-------
 2 files changed, 14 insertions(+), 16 deletions(-)

diff --git a/fdts/stm32mp257f-dk-ca35tdcid-fw-config.dtsi b/fdts/stm32mp257f-dk-ca35tdcid-fw-config.dtsi
index 3eedf590ad..9b05f0095f 100644
--- a/fdts/stm32mp257f-dk-ca35tdcid-fw-config.dtsi
+++ b/fdts/stm32mp257f-dk-ca35tdcid-fw-config.dtsi
@@ -1,19 +1,19 @@
-// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
 /*
  * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
  */
 
 /*
  * STM32MP25 tf-a firmware config
- * Project : eco
- * Generated by XLmx tool version 2.2 - 6/1/2023 12:15:35 PM
+ * Project : open
+ * Generated by XLmx tool version 2.2 - 3/6/2024 11:20:07 AM
  */
 
 / {
 	dtb-registry {
 		soc_fw-config {
-			load-address = <0x0 0x81ff0000>;
-			max-size = <0x10000>;
+			load-address = <0x0 0x81fc0000>;
+			max-size = <0x40000>;
 		};
 		tos_fw {
 			load-address = <0x0 0x82000000>;
@@ -22,13 +22,13 @@
 	};
 
 	st-mem-firewall {
-		bl31_context: bl31-context@81ff0000 {
-			reg = <0x0 0x81ff0000 0x0 0x10000>;
-			st,protreg = <RISAFPROT(RISAF_REG_ID(7), RIF_CID1_BF, RIF_CID1_BF, RIF_CID1_BF, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+		bl31_context: bl31-context@81fc0000 {
+			reg = <0x0 0x81fc0000 0x0 0x40000>;
+			st,protreg = <RISAFPROT(RISAF_REG_ID(7), RIF_CID0_BF|RIF_CID1_BF, RIF_CID0_BF|RIF_CID1_BF, RIF_CID1_BF, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
 		};
 		op_tee: op-tee@82000000 {
 			reg = <0x0 0x82000000 0x0 0x2000000>;
-			st,protreg = <RISAFPROT(RISAF_REG_ID(8), RIF_CID1_BF, RIF_CID1_BF, 0, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+			st,protreg = <RISAFPROT(RISAF_REG_ID(8), RIF_CID0_BF|RIF_CID1_BF, RIF_CID0_BF|RIF_CID1_BF, 0, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
 		};
 	};
 };
diff --git a/fdts/stm32mp257f-dk-ca35tdcid-rcc.dtsi b/fdts/stm32mp257f-dk-ca35tdcid-rcc.dtsi
index c3d32e726d..e3ebc8275e 100644
--- a/fdts/stm32mp257f-dk-ca35tdcid-rcc.dtsi
+++ b/fdts/stm32mp257f-dk-ca35tdcid-rcc.dtsi
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
 /*
  * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
  * Author: Loic Pallardy loic.pallardy@foss.st.com for STMicroelectronics.
@@ -6,8 +6,8 @@
 
 /*
  * STM32MP25 Clock tree device tree configuration
- * Project : eco
- * Generated by XLmx tool version 2.2 - 3/8/2023 2:34:26 PM
+ * Project : open
+ * Generated by XLmx tool version 2.2 - 3/6/2024 11:20:05 AM
  */
 
 &clk_hse {
@@ -46,7 +46,7 @@
 		FLEXGEN_CFG(2, XBAR_SRC_PLL4, 0, 1)
 		FLEXGEN_CFG(4, XBAR_SRC_PLL4, 0, 3)
 		FLEXGEN_CFG(5, XBAR_SRC_PLL4, 0, 2)
-		FLEXGEN_CFG(8, XBAR_SRC_PLL4, 0, 11)
+		FLEXGEN_CFG(8, XBAR_SRC_HSI_KER, 0, 0)
 		FLEXGEN_CFG(48, XBAR_SRC_PLL5, 0, 3)
 		FLEXGEN_CFG(51, XBAR_SRC_PLL4, 0, 5)
 		FLEXGEN_CFG(52, XBAR_SRC_PLL4, 0, 5)
@@ -68,14 +68,12 @@
 		};
 	};
 
-	/* DRAM clock = 2 * PLL2 clock */
 	pll2: st,pll-2 {
 		st,pll = <&pll2_cfg_600Mhz>;
 
 		pll2_cfg_600Mhz: pll2-cfg-600Mhz {
-			cfg = <59 1 4 1>;
+			cfg = <30 1 1 2>;
 			src = <MUX_CFG(MUX_MUXSEL6, MUXSEL_HSE)>;
-			frac = <0xF3B8CA>;
 		};
 	};
 
-- 
2.53.0

